src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -355,6 +355,18 @@ public enum EvalMetricType
         [Argument(ArgumentType.Multiple, HelpText = "Parallel LightGBM Learning Algorithm", ShortName = "parag")]
         public ISupportParallel ParallelTrainer = new SingleTrainerFactory();
 
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Use GPU for training?", ShortName = "gpu")]
+        public bool UseGPU = false;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "OpenCL platform ID. Usually each GPU vendor exposes one OpenCL platform. -1 means the system-wide default platform. Only used when UseGPU is true.", ShortName = "gpu_platform_id")]
+        public int GPUPlatformId = -1;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "OpenCL device ID in the specified platform. Each GPU in the selected platform has a unique device ID. -1 means the default device in the selected platform. Note: only used when UseGPU is true.", ShortName = "gpu_device_id")]
+        public int GPUDeviceId = -1;
+
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Use double precision math on GPU? Note: only used when UseGPU is true.", ShortName = "gpu_use_dp")]
+        public bool GPUUseDoublePrecision = false;
```

Comment:
Maybe it's overkill, but I would prefer to have interface ILightGbmExecutionDevice (or something like that)  with two implementations one is LightGbmOnCpu with zero parameters and second LightGbmOnGpu with all current parameters, and interface method  SetupOption(Dictionary<string, object> options).

So here you can create ISupportLightGbmExecutionDeviceFactory derived from IComponentFactory<ILightGbmExecutionDevice>. Then needed you can create instance from factory with .CreateComponent, and use instance  SetupOption method in ToDictionary.

as example you can check IEarlyStoppingCriterionFactory in FastTree project.